### ISSUE TITLE
Ability to use Embargo functionality without Advanced Workflow

### ISSUE THREAD

**Issue by adrexia** [silverstripe-australia/advancedworkflow#187]
It would be nice to have embargo functionality as a separate module so it can be used without needing the rest of advanced workflow. Could be included in advanced workflow as a composer dependency.

**Comment by phptek** [comment]
@adrexia Have a squiz at https://github.com/silverstripe-scienceninjas/embargoexpiry @mparkhill just stripped it out. It's not very elegant but does the job.

**Comment by adrexia** [comment]
Yep, I was still around when he did that. :)

I think, long term it would be good to just have two separate modules so you could add advanced workflow without embargo, just embargo, or both. They are common client requirements, but they can be completely separated without too much effort, and its not clear that embargo is related to the bulk of advanced workflow's functionality. 

On the other hand, there _are_ parts of the functionality that can come into conflict. One such scenario is this one:
* a workflow is assigned to a page that removes my right to publish, I then assign an embargo date to that page, and save the draft. What is supposed to happen? I don't have the right to publish, but it's not clear that the page won't be published, or what exactly needs to happen next (I *think* the solution is to request approval to publish?). 

The main advantage to keeping them together would be to fix some of these strange logic issues with better UX.

**Comment by nyeholt** [comment]
Agreed that it could (should?) be spun off into its own module; the existence in the same module comes about more down to the original 2.4 implementation being more tightly coupled due to the specific client requirement that spawned the embargo functionality, but it's since been pared back and more generic. 

There are some weird cases that come up when applying workflow with embargo (hence the existence of "desired" publish/unpublish date fields that are set if workflow is applied). So there is a UI issue that needs to be resolved around some of these cases, and I'm not sure whether that's better served keeping things coupled or not.

**Comment by chillu** [comment]
The Terraformers team in Auckland has given this another crack: https://github.com/silverstripe-terraformers/silverstripe-embargo-expiry/. It's built on SS4, and doesn't bring in any `Workflow*` classes like the last module trying this (https://github.com/silverstripe-scienceninjas/embargoexpiry). I've added some thoughts on how we could move advancedworkflow to use this module instead of building in embargo/expiry at https://github.com/silverstripe-terraformers/silverstripe-embargo-expiry/issues/15
